### ISSUE TITLE
[Question] is yt-dlc dead or has met with a similar fate as that of yt-dl before dmca takedown?

### ISSUE THREAD

**Issue by Ashish0804** [blackjack4494/yt-dlc#289]
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

It seems @blackjack4494 excitement to maintain yt-dlc has watered down. No new Merges in past 20ish days, last release was more than a month ago. 

I don't blame blackjack4494, everyone gets busy irl. All i am asking is will this project will be actively maintained or should i revert back to yt-dl?

Thanks and a Merry Christmas in advance to all

**Comment by nicolaasjan** [comment]
Yeah...
I have the same question.

Maybe @pukkandan knows more about this?
He has [forked](https://github.com/pukkandan/yt-dlc) this repo and there is recent activity in the [merge-ytdl2](https://github.com/pukkandan/yt-dlc/tree/merge-ytdl2) branch of it.

**Comment by cpz3501** [comment]
At least yt-dlc is working fine on youtube at the moment, can't say the same about ytdl.

**Comment by thinkpad4** [comment]
I hope he continues to release versions of this, I like having 2 options to use, this or yt-dl

**Comment by pukkandan** [comment]
@nicolaasjan Nope, I dont know any more than anyone else. A few weeks ago, @blackjack4494 had mentioned in an issue that he was having constant network issues. Afaik, we havent heard from him since.

Most of my fork's branches (including https://github.com/pukkandan/yt-dlc/tree/merge-ytdl2) are for creating PRs to this repo. I do have plans to clean up my repo to make it more suitable for public use. But I wont be creating any releases, at least for the timebeing. I simply lack the know-how or the time to maintain a large project like that.

**Comment by nicolaasjan** [comment]
> 
> Most of my fork's branches (including merge-ytdl2) are for creating PRs to this repo. I do have plans to clean up my repo to make it more suitable for public use. But I wont be creating any releases, at least for the time being. I simply lack the know-how or the time to maintain a large project like that.

Thanks!

I do hope however that @blackjack4494 will keep maintaining this repo, when he's got things sorted out, or that we'll at least hear something from him...

**Comment by vxbinaca** [comment]
it works, and a month isn't too terrible in terms of a release.

**Comment by jbruchon** [comment]
@pukkandan Let me know if I can help in any way. It could be that @blackjack4494 is unavailable for the holidays, but if the lapse in maintenance is too long, something will have to be done to move things forward, especially since no one else (I think) has push access to this repo.

**Comment by pukkandan** [comment]
@nicolaasjan @jbboehr  Like promised, I have cleaned up my fork.  There are no releases yet though. Those willing to directly run from source code can try it out. It is up to date with the latest versions of both youtube-dl and yt-dlc. There are also a few changes/additional features. https://github.com/pukkandan/yt-dlc/#changes

I have experience neither in making python releases nor maintaining a project of this scale. If someone can help/teach me to set up Github Actions (since Travis is now limited) and to create the binaries (with proper version info etc), then I can start providing releases. Otherwise dont expect them any time soon.

**Comment by nicolaasjan** [comment]
Thank you very much for your effort!

As a matter of fact, I've compiled binaries (both Linux and Windows) from the [merge-ytdl2](https://github.com/pukkandan/yt-dlc/tree/merge-ytdl2) branch for my own use for quite some time now and they worked fine.
So I can from now on use your [master branch](https://github.com/pukkandan/yt-dlc)?

Unfortunately I can't help you with any GitHub stuff, because my knowledge is even more limited...
Maybe you can also ask for help at the [youtube-dl subreddit](https://reddit.com/r/youtubedl/new/)?

**Comment by CatTheHacker** [comment]
@pukkandan If you tell me what exactly you need to have in GitHub Actions, I can help you with that.

**Comment by samoht0** [comment]
@pukkandan I'm rpm-packaging youtube-dlc in Fedora copr and switched to your fork for now. Works fine so far.
Hint: version.py needs an updated version string.

**Comment by pukkandan** [comment]
> If you tell me what exactly you need to have in GitHub Actions, I can help you with that.

Honestly, no idea. I have no experience with Github Actions/Travis at all.
Would I be able to directly use this commit? https://github.com/ytdl-org/youtube-dl/commit/a8b31505edbf4d9608b2d964b936b6720dcbb40d

**Comment by CatTheHacker** [comment]
Yes, but that workflow is for testing and linting only.

**Comment by pukkandan** [comment]
@nicolaasjan

> So I can from now on use your [master branch](https://github.com/pukkandan/yt-dlc)?

Yes. But be aware of the [changes in default behaviour](https://github.com/pukkandan/yt-dlc#2021010502) from yt-dlc

> I took the liberty to mention this at Reddit:

No problem

**Comment by Vangelis66** [comment]
@pukkandan 

For setting up GH Actions to compile Windows binaries, please take a look in: 

https://github.com/blackjack4494/yt-dlc/tree/master/.github/workflows

which contains 
https://github.com/blackjack4494/yt-dlc/blob/master/.github/workflows/build.yml
&
https://github.com/blackjack4494/yt-dlc/blob/master/.github/workflows/python-publish.yml.disable

The **x86 build** is compiled with Python 3.**4.4**, to be able to run on Windows XP SP3 32-bit and higher (Vista SP2 32/64 users, like myself, should use this; for my personal usage though, I'm compiling with Python 3.**7.9** - thanks @nicolaasjan 👍 )

The **x64 build** is compiled with Python 3.**8.7** to be able to run on Win7 SP1 64-bit and higher...

Your altruism and overall efforts are highly appreciated!

Kind regards!

**Comment by jbruchon** [comment]
@pukkandan I would recommend that, before you go any further, you decouple your fork from this fork. If something stupid happens to the parent repo like the youtube-dl takedown from a couple of months ago, your repo will automatically get nuked too.

**Comment by Vangelis66** [comment]
To make things even more perplexed, the **original**, DMCA'ed, repository of the application, 

https://github.com/blackjack4494/youtube-dlc

has been recently **reinstated** by GitHub, where people now continue to post **New Issues** there: 

https://github.com/blackjack4494/youtube-dlc/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

Should the original repo be made **read-only** and point visitors to here,

https://github.com/blackjack4494/yt-dlc

or should the current (second) repo be merged with the original one?
What a fine mess... 😄

**Comment by pukkandan** [comment]
I have made a discussion in my repo to discuss making a release for it. I dont want to pollute this thread with the discussion about the fork any more than we already have. 
@CatTheHacker @Vangelis66 I would appreciate if you could hop on there and help me set up the workflow
https://github.com/pukkandan/yt-dlc/discussions/1

**Comment by shakeyourbunny** [comment]
> 
> 
> @pukkandan I'm rpm-packaging youtube-dlc in Fedora copr and switched to your fork for now. Works fine so far.
> Hint: version.py needs an updated version string.

How do I enable this with CentOS 8?

I already installed the yum copr plugin and then I ran "yum copr enable samoht0/somerpmbuilds".

After that, it bailed out with
Do you really want to enable copr.fedorainfracloud.org/samoht0/somerpmbuilds? [y/N]: y
Error: This repository does not have any builds yet so you cannot enable it now.

**Comment by samoht0** [comment]
> How do I enable this with CentOS 8?

I'm just building for stable Fedora releases, as this is what I'm running and testing.

That said, the F32 rpms should work for this package in CentOS 8. You can download them and install / update the package manually
dnf install or upgrade "/path/to/rpm/youtube-dlc-...fc32.noarch.rpm"

The rpms are here:
https://copr-be.cloud.fedoraproject.org/results/samoht0/somerpmbuilds/fedora-32-x86_64/

Automatic package update won't work for obvious reason, bash competition *might* be broken (just test it).

**Comment by shakeyourbunny** [comment]
> 
> 
> > How do I enable this with CentOS 8?
> 
> I'm just building for stable Fedora releases, as this is what I'm running and testing.

My issue is that the copr repo doesn't integrate itself with it and bails out with this error message, so your COPR has an issue.

Installing gping ( yum copr enable atim/gping ) does work without a hitch.
